src/Ninject.Web.AspNetCore.Test/Unit/ServiceProviderTest.cs

```diff
@@ -0,0 +1,147 @@
+﻿using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Ninject.Web.AspNetCore.Test.Fakes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Ninject.Web.AspNetCore.Test.Unit
+{
+	public class ServiceProviderTest
+	{
+				
+		[Fact]
+		public void ServiceProviderCreatedSuccessfully()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			NinjectServiceProviderBuilder builder = CreateServiceProviderBuilder(kernel);
+
+			var provider = builder.Build();
+			
+			provider.Should().NotBeNull().And.BeOfType(typeof(NinjectServiceProvider));
+		}		
+
+		[Fact]
+		public void OptionalExistingSingleServiceResolved()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			var provider = CreateServiceProvider(kernel);
+			
+			provider.GetService(typeof(IWarrior)).Should().NotBeNull().And.BeOfType(typeof(Samurai));
+		}
+
+		[Fact]
+		public void OptionalNonExistingSingleServiceResolvedToNull()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			var provider = CreateServiceProvider(kernel);
+			
+			provider.GetService(typeof(IWarrior)).Should().BeNull();
+		}
+
+		[Fact]
+		public void OptionalExistingMultipleServicesResolvedQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+			
+			var result = provider.GetService(typeof(IList<IWarrior>)) as IEnumerable<IWarrior>;
+			
+			result.Should().NotBeNull();
+			var resultList = result.ToList();
+			resultList.Should().HaveCount(2);
+			resultList.Should().Contain(x => x is Samurai);
+			resultList.Should().Contain(x => x is Ninja);
+		}
+
+		[Fact]
+		public void OptionalExistingMultipleServicesNotResolvedWhenNotQueriedAsList()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			kernel.Bind<IWarrior>().ToConstant(new Ninja("test"));
+			var provider = CreateServiceProvider(kernel);
+
+			provider.GetService(typeof(IWarrior)).Should().BeNull();			
+		}
+
+		[Fact]
+		public void RequiredExistingSingleServiceResolved()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			kernel.Bind<IWarrior>().To<Samurai>();
+			var provider = CreateServiceProvider(kernel);
+
+			provider.GetRequiredService(typeof(IWarrior)).Should().NotBeNull().And.BeOfType(typeof(Samurai));
+		}
+
+		[Fact]
+		public void RequiredNonExistingSingleServiceResolvedToException()
+		{
+			StandardKernel kernel = CreateTestKernel();
+			var provider = CreateServiceProvider(kernel);
+
+			Action action = () => provider.GetRequiredService(typeof(IWarrior));
+			action.Should().Throw<ActivationException>().WithMessage("*No matching bindings are available*");
```

Comment:
If we don't implement ISupportRequiredService we would just get a InvalidOperationException without any nice explanation like in Ninject. For the not bound case, that would be fine but check the multiple matches cases...